components/list/demo/custom-nav-item.js

```diff
@@ -0,0 +1,87 @@
+import '../../colors/colors.js';
+import { css, LitElement } from 'lit';
+import { ListItemButtonMixin } from '../list-item-button-mixin.js';
+
+/**
+ * A component for a "listitem" child within a list. It provides semantics, basic layout, breakpoints for responsiveness, a link for navigation, and selection.
+ * @slot - Default content placed inside of the component
+ * @slot illustration - Image associated with the list item located at the left of the item
+ * @slot actions - Actions (e.g., button icons) associated with the listen item located at the right of the item
+ * @slot nested - Nested d2l-list element
+ */
+class ListItemCustomNavItem extends ListItemButtonMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * possible values: page, location
+			 */
+			current: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+
+		const styles = [ css`
+			:host([current="page"]) [slot="outside-control-container"] {
+				margin-block: 1px;
+				outline: 3px solid var(--d2l-button-focus-color, var(--d2l-color-celestine));
+			}
+			:host([current="location"]) [slot="outside-control-container"] {
+				margin-block: 1px;
+				outline: 3px solid var(--d2l-button-focus-color, var(--d2l-color-celestine-plus-2));
```

Comment:
This is just to make it easy to see when an item has "location"; wouldn't be used in practice